backends/authority/organization.go

```diff
@@ -0,0 +1,166 @@
+package authority
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/elastic/go-elasticsearch/v6/esapi"
+	"github.com/ugent-library/biblio-backoffice/models"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type organizationSearchEnvelope struct {
+	Hits struct {
+		Total int `json:"total"`
+		Hits  []struct {
+			ID     string              `json:"_id"`
+			Source models.Organization `json:"_source"`
+		} `json:"hits"`
+	} `json:"hits"`
+}
+
+type searchEnvelope struct {
+	Hits struct {
+		Total int `json:"total"`
+		Hits  []struct {
+			ID     string          `json:"_id"`
+			Source json.RawMessage `json:"_source"`
+		} `json:"hits"`
+	} `json:"hits"`
+}
+
+//go:embed organization_settings.json
+var organizationSettings string
+
+func (c *Client) EnsureOrganizationSeedIndexExists() error {
+	res, err := esapi.IndicesExistsRequest{
+		Index: []string{"biblio_organization"},
+	}.Do(context.Background(), c.es)
+	if err != nil {
+		return err
+	}
+	if res.StatusCode == 404 {
+		res, err := c.es.Indices.Create("biblio_organization", c.es.Indices.Create.WithBody(strings.NewReader(organizationSettings)))
+		if err != nil {
+			return err
+		}
+		if res.IsError() {
+			return fmt.Errorf("%+v", res)
+		}
+		time.Sleep(5 * time.Second)
+	}
+	return nil
+}
+
+func (c *Client) SeedOrganization(data []byte) error {
```

Comment:
Why do you provide a raw byte array instead of an Organization

Reply:
see comment below